rb/spec/integration/selenium/webdriver/network_spec.rb

```diff
@@ -262,6 +262,20 @@ module WebDriver
         end
       end
 
+      it 'adds a response handler that provides a response' do
+        reset_driver!(web_socket_url: true) do |driver|
+          network = described_class.new(driver)
+          network.add_response_handler do |response|
+            response.status = 200
+            response.headers['foo'] = 'bar'
+            response.body = '<html><head><title>Hello World!</title></head><body/></html>'
+            response.provide_response
+          end
+          driver.navigate.to url_for('formPage.html')
+          expect(driver.page_source).to include('Hello World!')
```

Comment:
Is it worth for these tests to also verbosely include that the page isn't returned, i.e. that page (no idea what it includes), doesn't show it's text?

Reply:
So the method's idea is that once the page is intercepted, it will return 'Hello World!' as it's content since we are providing a new response

This is why we only check for 'Hello World!'

Reply:
Yes I'm aware. What I'm saying is, as a proof of concept this is good, but what would be more robust is assertions that we "haven't" got stuff from the `formPage.html` - text / titles /

Reply:
Obviously the page url wouldn't be a good assertion, but some page content not showing **would** be a good assertion